ksql-engine/src/main/java/io/confluent/ksql/function/udaf/KudafAggregator.java

```diff
@@ -50,22 +51,54 @@ public GenericRow apply(final Struct k, final GenericRow rowValue, final Generic
     // compute the aggregation and write it into the output row. Its assumed that
     // the columns written by this statement do not overlap with those written by
     // the above statement.
-    aggValToAggFunctionMap.forEach((key, value) ->
-        aggRowValue.getColumns().set(
-            key,
-            value.aggregate(
-                rowValue.getColumns().get(value.getArgIndexInValue()),
-                aggRowValue.getColumns().get(key))));
+    for (Entry<Integer, KsqlAggregateFunction> entry: aggValToAggFunctionMap.entrySet()) {
+      final int index = entry.getKey();
+      final KsqlAggregateFunction ksqlAggregateFunction = entry.getValue();
+      aggRowValue.getColumns().set(
+            index,
+            ksqlAggregateFunction.aggregate(
+                rowValue.getColumns().get(ksqlAggregateFunction.getArgIndexInValue()),
+                aggRowValue.getColumns().get(index)));
+    }
 
     return aggRowValue;
   }
 
+  @SuppressWarnings("unchecked")
+  public ValueMapper<GenericRow, GenericRow> getResultMapper() {
+
+    return (aggRow) -> {
+      final int size = aggRow.getColumns().size();
+      final List<Object> columns = new ArrayList<>(size);
+      for (int i = 0; i < size; i++) {
```

Comment:
do we need this initialization loop? If so, let's extract a constant for `""` and call it `UNINITIALIZED_COLUMN` to make it clear what this is doing

Reply:
I see that you changed it to `UNINITIALIZED_COLUMN`, but I'm wondering still why we need it at all? Can you add a comment to why we need that in the code?